model/src/main/kotlin/ScannerRun.kt

```diff
@@ -273,6 +273,50 @@ data class ScannerRun(
         scanResultsById.mapValues { (_, scanResults) ->
             scanResults.flatMapTo(mutableSetOf()) { it.summary.issues }
         }.zipWithSets(issues)
+
+    operator fun plus(other: ScannerRun): ScannerRun {
+        require(environment == other.environment) {
+            "Cannot merge ScannerRuns with different environments: $environment != ${other.environment}."
+        }
+
+        require(config == other.config) {
+            "Cannot merge ScannerRuns with different configurations: $config != ${other.config}."
+        }
+
+        val mergedFiles = (files.toList() + other.files.toList())
+            .groupBy { it.provenance }
+            .values
+            .map { fileLists ->
```

Comment:
Prefer `mapTo(mutableSet())` to avoid a separate `toSet()` conversion in the end.